File: trunk/zubat/src/com/k42b3/zubat/Configuration.java
/**
 * zubat
 * 
 * An java application to access the API of amun. It is used to debug and
 * control a website based on amun. This is the reference implementation 
 * howto access the api. So feel free to hack and extend.
 * 
 * Copyright (c) 2011 Christoph Kappestein <k42b3.x@gmail.com>
 * 
 * This file is part of oat. oat is free software: you can 
 * redistribute it and/or modify it under the terms of the GNU 
 * General Public License as published by the Free Software Foundation, 
 * either version 3 of the License, or at any later version.
 * 
 * oat is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with oat. If not, see <http://www.gnu.org/licenses/>.
 */

package com.k42b3.zubat;

import java.io.File;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

/**
 * Configuration
 *
 * @author     Christoph Kappestein <k42b3.x@gmail.com>
 * @license    http://www.gnu.org/licenses/gpl.html GPLv3
 * @link       http://code.google.com/p/delta-quadrant
 * @version    $Revision$
 */
public class Configuration 
{
	private String baseUrl;
	private String consumerKey;
	private String consumerSecret;
	private String token;
	private String tokenSecret;

	public String getBaseUrl() 
	{
		return baseUrl;
	}

	public void setBaseUrl(String baseUrl) 
	{
		this.baseUrl = baseUrl;
	}

	public String getConsumerKey() 
	{
		return consumerKey;
	}

	public void setConsumerKey(String consumerKey) 
	{
		this.consumerKey = consumerKey;
	}

	public String getConsumerSecret() 
	{
		return consumerSecret;
	}

	public void setConsumerSecret(String consumerSecret) 
	{
		this.consumerSecret = consumerSecret;
	}

	public String getToken() 
	{
		return token;
	}

	public void setToken(String token) 
	{
		this.token = token;
	}

	public String getTokenSecret() 
	{
		return tokenSecret;
	}

	public void setTokenSecret(String tokenSecret) 
	{
		this.tokenSecret = tokenSecret;
	}

	public static Configuration parseFile(File configFile) throws Exception
	{
		Configuration config = new Configuration();

		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
		DocumentBuilder db = dbf.newDocumentBuilder();
		Document doc = db.parse(configFile);

		Element rootElement = (Element) doc.getDocumentElement();

		rootElement.normalize();


		// parse config elements
		String baseUrl = "";
		String consumerKey = "";
		String consumerSecret = "";
		String token = "";
		String tokenSecret = "";

		Element baseUrlElement = (Element) doc.getElementsByTagName("baseUrl").item(0);
		Element consumerKeyElement = (Element) doc.getElementsByTagName("consumerKey").item(0);
		Element consumerSecretElement = (Element) doc.getElementsByTagName("consumerSecret").item(0);
		Element tokenElement = (Element) doc.getElementsByTagName("token").item(0);
		Element tokenSecretElement = (Element) doc.getElementsByTagName("tokenSecret").item(0);
		
		if(baseUrlElement != null)
		{
			baseUrl = baseUrlElement.getTextContent();
		}
		else
		{
			throw new Exception("baseUrl in config not set");
		}
		
		if(consumerKeyElement != null)
		{
			consumerKey = consumerKeyElement.getTextContent();
		}
		else
		{
			throw new Exception("consumerKey in config not set");
		}
		
		if(consumerSecretElement != null)
		{
			consumerSecret = consumerSecretElement.getTextContent();
		}
		else
		{
			throw new Exception("consumerSecret in config not set");
		}

		if(tokenElement != null)
		{
			token = tokenElement.getTextContent();
		}

		if(tokenSecretElement != null)
		{
			tokenSecret = tokenSecretElement.getTextContent();
		}

		config.setBaseUrl(baseUrl);
		config.setConsumerKey(consumerKey);
		config.setConsumerSecret(consumerSecret);
		config.setToken(token);
		config.setTokenSecret(tokenSecret);

		return config;
	}
}

File: metang/configuration.php
<?php
/*
 *  $Id: configuration.php 270 2011-04-16 19:30:28Z k42b3.x $
 *
 * psx
 * A object oriented and modular based PHP framework for developing
 * dynamic web applications. For the current version and informations
 * visit <http://phpsx.org>
 *
 * Copyright (c) 2011 Christoph Kappestein <k42b3.x@gmail.com>
 *
 * This file is part of psx. psx is free software: you can
 * redistribute it and/or modify it under the terms of the
 * GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or any later version.
 *
 * psx is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with psx. If not, see <http://www.gnu.org/licenses/>.
 */

$config = array(

	'metang_version'          => '0.0.1 beta',
	'metang_url'              => 'http://127.0.0.1/projects/amun/public/index.php/',

	'metang_consumer_key'     => 'b8858501073e5fb54e75b973ed044ec19f21a60d',
	'metang_consumer_secret'  => '07d8b5173afba2575e57ca5966624a39419b5b70',

	'metang_request'          => 'http://127.0.0.1/projects/amun/public/index.php/api/auth/request',
	'metang_authorization'    => 'http://127.0.0.1/projects/amun/public/index.php/api/auth/authorization',
	'metang_access'           => 'http://127.0.0.1/projects/amun/public/index.php/api/auth/access',

	'psx_version'             => '0.3.3',
	'psx_url'                 => 'http://127.0.0.1/projects/metang/public',
	'psx_dispatch'            => 'index.php/',
	'psx_timezone'            => 'Europe/Berlin',
	'psx_gzip'                => false,
	'psx_debug'               => true,

	'psx_module_default'      => 'main',
	'psx_module_input'        => isset($_GET['x']) ? $_GET['x'] : (isset($_SERVER['PATH_INFO']) ? $_SERVER['PATH_INFO'] : (isset($_SERVER['argv'][1]) ? $_SERVER['argv'][1] : '')),
	'psx_module_input_length' => 256,

	'psx_sql_host'            => 'localhost',
	'psx_sql_user'            => 'root',
	'psx_sql_pw'              => '',
	'psx_sql_db'              => 'psx',

	'psx_cache_enabled'       => true,
	'psx_cache_expire'        => 60 * 30,

	'psx_template_dir'        => 'default',
	'psx_template_default'    => false,

	'psx_path_cache'          => '../cache',
	'psx_path_library'        => '../library',
	'psx_path_module'         => '../module',
	'psx_path_template'       => '../template',

);

